proto/api/v1/authprovider_service.proto

```diff
@@ -111,10 +119,10 @@ service AuthProviderService {
         };
     }
 
-    rpc PutAuthProvider (storage.AuthProvider) returns (storage.AuthProvider) {
+    rpc PutAuthProvider (PutAuthProviderRequest) returns (storage.AuthProvider) {
```

Comment:
Note that this change is _not_ gRPC wire compatible! IMHO this is okay because we do not provide compatibility guarantees, and there are no gRPC customers of this method that we know of. However, in the future we might need to be more careful. One take-away would be that we should maybe officially disallow `storage.` types as input for service methods (i.e., _always_ follow the scheme `rpc MyRPCCall (MyRPCCallRequest) returns (MyRPCCallResponse)`, with an exception for common types such as `DeleteByID` and `Empty`).

However, (and I feel bad to only point this out now, after all those changes, but better late than never) I don't feel it was necessary to add a `force` parameter to the request. My comment https://github.com/stackrox/stackrox/pull/2663#discussion_r945588805 was about the following comment:
```
// We do not allow updates to groups even with the force flag set, hence explicitly setting force to false here.
```
This suggests that `force` is ignored, and it was magically being set to `false`. However, I do not see an issue with omitting `force` from the API. The point is that the mutability enforcement happens at the datastore, and there it would be good to implement a consistent logic. But if the service handler decides to not offer a `force` parameter, that is just an API choice - there is no logic inconsistency in `ALLOW_MUTATE_FORCED` suddenly not meaning what the word says.

I admit that I might have derailed the discussion a bit by questioning the motivation of why allow delete but not mutate _via API_, but my main comment was regarding the implementation at the datastore level, and I don't think it's a given that this needs to impact the API.

Reply:
I don't think we should break API, as we don't do it in other places, e.g., [group service](https://github.com/stackrox/stackrox/pull/2663/files#diff-fe7c26168cea62173cff7ce47d14c3653edf35583ebb3ecd62f3ce4ff2da9f9bR50-R51). We indeed don't provide guarantees but there is some chance that someone's script is broken for no good reason.

Reply:
Was aware that it wasn't gRPC wire compatible, but I haven't found any references to that specific function and the HTTP API was compatible.

Definitely +1 on forbidding `storage.*` as input.

However, I suppose I'll just revert this then, and stick with ignoring the force flag for update then.